test/framework/src/main/java/org/opensearch/test/OpenSearchIntegTestCase.java

```diff
@@ -2622,4 +2631,72 @@ protected ClusterState getClusterState() {
         return client(internalCluster().getClusterManagerName()).admin().cluster().prepareState().get().getState();
     }
 
+    /**
+     * Checks if replica shards caught up with primary shard when Segment Replication is enabled.
+     */
+    protected void verifyReplicasCaughtUpWithPrimary() {
```

Comment:
We have methods to do this wait already in segrep IT, can we move those here and reuse?

Reply:
Done moved to OpenSearchIntegTestCase

Reply:
👍